src/Perlang.Interpreter/PerlangInterpreter.cs

```diff
@@ -1226,9 +1227,22 @@ public VoidObject VisitWhileStmt(Stmt.While stmt)
                 {
                     return Convert.ToDouble(value);
                 }
+                else if (targetTypeReference.ClrType == typeof(BigInteger))
+                {
+                    return value switch
+                    {
+                        int intValue => new BigInteger(intValue),
+                        long longValue => new BigInteger(longValue),
+                        float floatValue => new BigInteger(floatValue),
+                        double doubleValue => new BigInteger(doubleValue),
+
+                        // TODO: Might need to revisit this to support more types as we implement #70.
+                        _ => throw new IllegalStateException($"Unsupported conversion from {value.GetType().ToTypeKeyword()} to {targetTypeReference.ClrType.ToTypeKeyword()}")
+                    };
+                }
```

Comment:
:point_up: This is the essence of the fix.

Also, the TODO comment for #70 is quite important. The easiest way to ensure that it gets fixed is probably to add more test cases, so that we properly test this with values of all types expected to be supported (i.e. expanding this list as we support more types, `uint`, `ulong` and potentially `decimal` - there are `BigInteger` constructors for all of these that could be used).